src/runtime/ObjectStructure.h

```diff
@@ -149,13 +149,16 @@ class ObjectStructure : public gc {
 
     virtual bool inTransitionMode() = 0;
     virtual bool hasIndexPropertyName() = 0;
+    virtual bool hasSymbolPropertyName() = 0;
     virtual bool hasEnumerableProperty() = 0;
 };
 
 class ObjectStructureWithoutTransition : public ObjectStructure {
 public:
-    ObjectStructureWithoutTransition(ObjectStructureItemVector* properties, bool hasIndexPropertyName, bool hasNonAtomicPropertyName, bool hasEnumerableProperty)
+    ObjectStructureWithoutTransition(ObjectStructureItemVector* properties, bool hasIndexPropertyName,
+                                     bool hasSymbolPropertyName, bool hasNonAtomicPropertyName, bool hasEnumerableProperty)
         : m_hasIndexPropertyName(hasIndexPropertyName)
+        , m_hasSymbolPropertyName(hasSymbolPropertyName)
         , m_hasNonAtomicPropertyName(hasNonAtomicPropertyName)
         , m_hasEnumerableProperty(hasEnumerableProperty)
         , m_properties(properties)
```

Comment:
It is not about this patch, 
but I'm  just wondering that `ObjectStructure` classes could share several member variables like above but these are defined in each class.
Is there any special reason for this?
How about moving these variables into upper `ObjectStructure` class as a kind of refactoring?

Reply:
Good catch!

Reply:
I've updated my pr